cs/src/Management/TunnelManagementClient.cs

```diff
@@ -519,6 +519,15 @@ private string UserLimitsPath
 
             request.Headers.UserAgent.Add(TunnelSdkUserAgent);
 
+            const string regKeyPath = @"Software\Policies\Microsoft\Tunnels";
+            var defaultOnError = true;
+            RegistryTools policies = new RegistryTools();
+            var keyValue = (string)policies.GetRegistryValueFromLocalMachineRoot(regKeyPath, defaultOnError);
```

Comment:
We can extract this piece of code in a separate method such that there is a policyProvider object that returns list of policies and policy values. PolicyProvider class can have the registry path set and intialize RegistryTools and it calls getregistryValue that you have specified in another file.